solr/core/src/java/org/apache/solr/handler/admin/ClusterStatus.java

```diff
@@ -109,7 +109,7 @@ public void getClusterStatus(NamedList<Object> results)
     boolean withClusterProperties = solrParams.getBool(CLUSTER_PROP, includeAll);
     boolean withRoles = solrParams.getBool(ZkStateReader.ROLES_PROP, includeAll);
     boolean withCollection = includeAll || (collection != null);
-    boolean withAliases = solrParams.getBool(ALIASES_PROP, includeAll);
+    boolean withAliases = solrParams.getBool(ALIASES_PROP, includeAll) || withCollection;
```

Comment:
but if the user specifies "aliases=false" why overrule their decision?

Reply:
I agree - this conflicts with your comment here https://github.com/apache/solr/pull/2599/files#r1707539294

The `withAliases` refers to the property which when enabled would fetch the entire aliasMap stored in memory. 

When `collection` parameter is non-null, aliases are fetched from ZkStateReader to build the mapping from collections to aliases or vice versa when the input is an alias.

It then makes sense to keep this property separate as:
`boolean withAliases = solrParams.getBool(ALIASES_PROP, includeAll);`

I am not sure what your other comment meant.

Reply:
Updated the PR to request for aliases explicitly